### ISSUE TITLE
tf.summary.FileWriter crashes with AlreadyExistsError

### ISSUE THREAD

**Issue by una-dinosauria** [tensorflow/tensorflow#7448]
We have a cluster with ~20 GPUs that we often use to train multiple networks in parallel, and we use `tf.summary.Filewriter` to keep track of the networks' progress. However, some jobs are crashing when they attempt to create their `FileWriter`s with the following stack trace:

```
[...]
  File "/ubc/cs/research/tracking-raid/julm/eyescream/tensorflow/pose_estimation/linear_model.py", line 141, in __init__
    self.train_writer = tf.summary.FileWriter( os.path.join(summaries_dir, 'train' ))
  File "/ubc/cs/research/tracking-raid/julm/anaconda/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/summary/writer/writer.py", line 308, in __init__
    event_writer = EventFileWriter(logdir, max_queue, flush_secs)
  File "/ubc/cs/research/tracking-raid/julm/anaconda/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/summary/writer/event_file_writer.py", line 69, in __init__
    gfile.MakeDirs(self._logdir)
  File "/ubc/cs/research/tracking-raid/julm/anaconda/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/lib/io/file_io.py", line 299, in recursive_create_dir
    pywrap_tensorflow.RecursivelyCreateDir(compat.as_bytes(dirname), status)
  File "/ubc/cs/research/tracking-raid/julm/anaconda/envs/tensorflow/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/ubc/cs/research/tracking-raid/julm/anaconda/envs/tensorflow/lib/python2.7/site-packages/tensorflow/python/framework/errors_impl.py", line 469, in raise_exception_on_not_ok_status
    pywrap_tensorflow.TF_GetCode(status))
tensorflow.python.framework.errors_impl.AlreadyExistsError: /global/scratch/julm/3d_experiments
```

I am writing the `Filewriter`s' progress under `/global/scratch/julm/3d_experiments/`, and creating multiple subdirectories depending on the hyperparameters that each network is using.

The error seems to suggest that the `FileWriter` is trying to create the directory `/global/scratch/julm/3d_experiments/` and crashing because the directory already exists. Moreover, only around 1 in 5 jobs crashes with this error.

Do you know if I could somehow ignore this error? I don't think the fact that the directory exists should trigger an error for the user.

Our cluster runs under OpenSUSE 42.2.

**Comment by aselle** [comment]
I think you have a race condition. If you start /global/scratch/julm/3d_experiments/ before starting any experiments, does it create any errors. What happens otherwise is the first process looks and doesn't see it and then between the time it looked somebody else looked and created it.

**Comment by jsseely** [comment]
I also get this race condition. The cluster has Red Hat Enterprise Linux 7 and Slurm job scheduler. Using TF 0.12. Creating the directory before submitting the job fixes the problem. But, handling the race condition within `FileWriter` might be beneficial?

**Comment by una-dinosauria** [comment]
Running
```python
os.system('mkdir -p {}'.format( summaries_directory ))
```
does get rid of the race condition -- or, rather, let's linux handle the race condition, and that fixes my particular problem.

I agree with @aselle that it'd be super nice to have TF handle this internally though.

**Comment by aselle** [comment]
@una-dinosauria, it's unlikely that this will be done by us in the short term or long term, but feel free to contribute a PR to TensorFlow. Thanks!

**Comment by una-dinosauria** [comment]
@aselle Thanks! I'd love to contribute to TF, and this sounds like it has a fairly straightforward fix. 
It's unlikely that I'll find the time to do it before mid-March though -- I guess we should keep this open just to track the issue and as a reference for others who might run into this problem.

**Comment by dksb** [comment]
Closing due to lack of recent activity. Please reopen and update the issue if you submit a PR for the change. Thanks!
